dev/remove-unused-images.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+  echo "Usage: $0 [--check]"
+  echo "  --check  List unused images without deleting them"
+}
+
+MODE="delete"
+if [[ $# -gt 1 ]]; then
+  usage
+  exit 1
+fi
+if [[ $# -eq 1 ]]; then
+  case "$1" in
+    --check|-c) MODE="check" ;;
+    *) usage; exit 1 ;;
+  esac
+fi
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd "$REPO_ROOT"
+
+IMAGE_EXTS=(png jpg jpeg gif)
+IMAGE_PATTERNS=()
+for ext in "${IMAGE_EXTS[@]}"; do
+  IMAGE_PATTERNS+=("docs/**/*.${ext}")
+done
+
+IMAGES=()
+while IFS= read -r -d '' file; do
+  IMAGES+=("$file")
+done < <(git ls-files -z -- "${IMAGE_PATTERNS[@]}")
+
+if [[ ${#IMAGES[@]} -eq 0 ]]; then
+  echo "No tracked image files found."
+  exit 0
+fi
+
+SEARCH_FILES=()
+while IFS= read -r -d '' file; do
+  SEARCH_FILES+=("$file")
+done < <(git ls-files -z docs mlflow || true)
+if [[ ${#SEARCH_FILES[@]} -eq 0 ]]; then
+  echo "No tracked files under docs/ or mlflow/ to search for references."
+  exit 0
+fi
+
+if ! command -v rg >/dev/null 2>&1; then
+  echo "Error: ripgrep (rg) is required but not found. Install it with: brew install ripgrep"
+  exit 1
+fi
+
+search_text() {
+  # Search only tracked docs and mlflow files to avoid matches in ignored/untracked content.
+  printf '%s\0' "${SEARCH_FILES[@]}" | xargs -0 rg --no-messages --fixed-strings --quiet --files-with-matches -- "$1" >/dev/null
```

Comment:
grep is slow and behaves differently on macOS and Linux. ripgrep (rg) is fast and consistent across platforms (but not pre installed in GA).